### DISCUSSION TITLE
Add tags to content when saving

### DISCUSSION THREAD

**Discussion by xave** [gildas-lormeau/SingleFile#1340] [Q&A]
Add tags to content when saving

I’m looking to add tags to a file when saving. What’s the best workflow to this. Ideally, the information would not change the content itself, so I’m thinking the title of the file document itself. An ideal solution would add the tags to metadata.

**Comment by gildas-lormeau** [comment]
The tags would we stored in a good old `<meta name="keywords" content="...">` tag in the saved page?

**Comment by xave** [comment]
Thanks for clarifying. In my mind, the entire html document would be as is normally made with the tool. Yet at the point of saving a person could annotate that document either in the title or in perhaps `xattr` metadata for the file itself. 

The use case here is similar to Evernote web clipper. You have your tags database and when you save a file you can add tags from there to it. Since SingleFile is not locking one into a proprietary format, it would seem that either adding the tags to the filename string or to an `xattr` or similar would accomplish a similar effect. 

The ultimate goal is to have some system of organizing downloaded pages along the way at the moment of creation when one's sense of what a document is about is at its most fresh. I guess a person could add those tags externally outside of this tool. Though it does seem to be the type of functionality that could fit into the "annotation" aspect of the settings or similar.

**Reply by xave** [reply]
To clarify further, I'm referring to  a distinction between the tags already in a webpage and a person's personal schema for organizing and tagging information.

**Reply by gildas-lormeau** [reply]
I will redact an answer to your question. However, out of curiosity, did you try to use Zotero? Its "snapshot" feature relies on SingleFile under the hood. There's also https://github.com/nuthrash/obsidian-html-plugin for Obsidian and https://github.com/xnohat/trilium-simple-singlefile-renderer for Trillium, FYI.

**Reply by xave** [reply]
Thank you for your thoughtful response. I am aware of those other options. I tend to manage my files at the OS level.
